### ISSUE TITLE
Windows GPU not working.  CUDNN_STATUS_BAD_PARAM

### ISSUE THREAD

**Issue by strickon** [thtrieu/darkflow#98]
So I have been trying to run on windows 10 , gtx 1060, cudnn 5.1 cuda 8, and tensorflow-gpu and keep getting these errors.  Tensforlow CPU works fine.

E c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_dnn.cc:397] could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
E c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_dnn.cc:404] error retrieving driver version: Permission denied: could not open driver version path for reading: /proc/driver/nvidia/version
E c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_dnn.cc:364] could not destroy cudnn handle: CUDNN_STATUS_BAD_PARAM
F c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\core\kernels\conv_ops.cc:605] Check failed: stream->parent()->GetConvolveAlgorithms(&algorithms) 

I did a bunch of digging and testing.  It seems that these errors stem from something related to conv2d and cudnn.  Unfortunately you can't turn cudnn off which I tried as there is no gpu version of conv2d that doesn't use it.  

I have also read that this error sometimes is masking another completely unrelated error.   Some say it has to do with empty numpy arrays.  I have been trying to track down this error.  I think the CPU version may be more lenient in not crashing on this error so that is why we see it run there.  

Anyone have any luck resolving this?

**Comment by strickon** [comment]
looks like if I adjust the gpu usage to .7 it works.  I am not understanding what that actually means and how that other error should manifest itself.

**Comment by thtrieu** [comment]
Although you did not post the error message (only traceback is provided) I think this is solely a memory problem. When one runs out of memory, error messages can be very weird.
Close since the original issue is resolved using `--gpu 0.7`

**Comment by uptown** [comment]
@strickon I don't know why, but there's a discussion similar with this issue. 
https://devtalk.nvidia.com/default/topic/901646/-980-ti-windows-10-cuda-7-5-out-of-memory-after-allocating-4-5-out-of-6gb/?offset=4
In short, avoiding to use Windows 10 is the best solution!

BTW, using `--gpu 0.85` worked for me in Windows 10.

**Comment by PaulHuynh489** [comment]
I'm running into the same issue.  How do you set --gpu 0.8?

**Comment by janakptl00** [comment]
Running into simmilar problem, would love some explanation on how to set ''--gpu" parameter ..

**Comment by Venka97** [comment]
@janakptl00  @PaulHuynh489 

```
config = tf.ConfigProto()
config.gpu_options.per_process_gpu_memory_fraction = 0.4
session = tf.Session(config=config, ...)
```

This should work.
Check this thread: https://github.com/tensorflow/tensorflow/issues/6698

**Comment by uptown** [comment]
hey guys, my computer has multiple gpus, so I can set --gpu 0.85 with "GPU NOT FOR DISPLAY" ( I thinks 0.15 loss is because of the Win10 issue).
If you use GPU which is also handling your display, you may reduce GPU usage rate.

**Comment by Vozf** [comment]
For me the problem was using wrong cudnn lib
I used cudnn for cuda 9.1 when I had cuda 9.0. So i reinstalled cudnn for cuda 9.0 and everything worked.

**Comment by zmchang** [comment]
Thanks very much.I fixed it by changing cudnn from 9 to cudnn 8 and it worked.H  ave you tried to use different models to run this tutorial?I have issues, when I tried to use inception resnet V2 model.
William

**Comment by Vozf** [comment]
Nope. I just found this issue by googling. I had this problem in my project using inceptionV3. 24.04.2018, 05:06, "zmchang" <notifications@github.com>:Thanks very much.I fixed it by changing cudnn from 9 to cudnn 8 and it worked.H  ave you tried to use different models to run this tutorial?I have issues, when I tried to use inception resnet V2 model.WilliamFrom: Alexander <notifications@github.com>To: thtrieu/darkflow <darkflow@noreply.github.com>Cc: zmchang <wzchang@sbcglobal.net>; Comment <comment@noreply.github.com>Sent: Sunday, April 22, 2018 10:08 AMSubject: Re: [thtrieu/darkflow] Windows GPU not working. CUDNN_STATUS_BAD_PARAM (#98)For me the problem was using wrong cudnn libI used cudnn for cuda 9.1 when I had cuda 9.0. So i reinstalled cudnn for cuda 9.0 and everything worked.—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread. —You are receiving this because you commented.Reply to this email directly, view it on GitHub, or mute the thread.
